ibis/backends/datafusion/__init__.py

```diff
@@ -0,0 +1,141 @@
+from __future__ import annotations
+
+import re
+from typing import Mapping
+
+import datafusion as df
+import pkg_resources
+import pyarrow as pa
+
+import ibis.common.exceptions as com
+import ibis.expr.schema as sch
+import ibis.expr.types as ir
+from ibis.backends.base import BaseBackend
+
+from .compiler import translate
+
+# TODO(kszucs): support nested and parametric types
+# consolidate with the logic from the parquet backend
+
+
+class Backend(BaseBackend):
+    name = 'datafusion'
+    builder = None
+
+    @property
+    def version(self):
+        dist = pkg_resources.get_distribution("datafusion")
```

Comment:
Use `importlib.metadata`/`importlib_metadata` for this. `pkg_resources` is slow and unnecessary at this point.

Reply:
probably can use `Backend.name` to load this as well.

Reply:
Updated.